### PR TITLE
Create actions to Add, edit, delete distribution configs in store

### PR DESCRIPTION
**Pull request by patjanawat**
Create actions to Add, edit, delete distribution configs in store (Closes)

Related https://github.com/mesg-foundation/atd/issues/51

### CODE REVIEW

File: frontend/store/allowance.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    commit('setTransactionHash','')
+
+    const latestContract = atd_contracts.find(atd => atd.to === 'latest').address 
+    const accounts = await (await artifacts).accounts() 
+    const token = (await artifacts).token
+   
+    const transaction = token.methods
+          .approve(
+            latestContract,
+            value.toString().replace(".","").padEnd(18,'0')
```
Comment (NicolasMahe) [inline frontend/store/allowance.js]: Don't use padEnd. Use BigNumber and a classic multiplication.

File: frontend/store/allowance.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          .on('transactionHash', hash => {            
+            commit('setTransactionHash',hash)
+          })
+           .on('confirmation', (confirmationNumber, receipt) => {
+            console.log('confirmationNumber', confirmationNumber, ':',receipt)            
+          })
+          .on('receipt', async (receipt) => {
+              console.log('receipt', receipt)              
+              const returnValues = receipt.events["Approval"].returnValues
+              console.log('receipt', returnValues)
```
Comment (NicolasMahe) [inline frontend/store/allowance.js]: can you remove all useless console.log?

File: frontend/store/distributionConfige.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          .on('transactionHash', hash => {            
+            commit('setTransactionHash',hash)            
+          })
+           .on('confirmation', (confirmationNumber, receipt) => {
+            // console.log('confirmationNumber', confirmationNumber, ':',receipt)            
+          })
+          .on('receipt', async (receipt) => {
+              // console.log('receipt', receipt)              
+              // const returnValues = receipt.events["DistributionConfigAdded"].returnValues
+              // console.log('receipt', returnValues)
```
Comment (NicolasMahe) [inline frontend/store/distributionConfige.js]: can you remove commented lines

File: frontend/store/distributionConfige.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import Artifacts from '../artifacts'
```
Comment (NicolasMahe) [inline frontend/store/distributionConfige.js]: the file name has a typo

File: frontend/store/distributionConfige.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    state.status = done
+  },
+  setTransactionHash:(state,hashValue) =>{
+    state.transactionHash = hashValue
+  }
+}
+
+export const state = () => ({
+  transactionHash:'', 
+  status:false
```
Comment (NicolasMahe) [inline frontend/store/distributionConfige.js]: can you use a linter and make sure to fix indentation, spaces, etc...

File: frontend/components/admin/allowance.vue [hunk 6]
```diff
@@ -1,105 +1,140 @@
...
-          <v-spacer></v-spacer>
-          <v-btn color="blue darken-1" flat @click="close">Cancel</v-btn>
-          <v-btn color="blue darken-1" flat @click="save">Save</v-btn>
-        </v-card-actions>
-      </v-card>
-    </v-dialog>
+          <v-card-actions>
+            <v-spacer></v-spacer>
+            <v-btn color="blue darken-1" flat @click="close">Cancel</v-btn>
+            <v-btn color="blue darken-1" flat @click="save">Save</v-btn>
```
Comment (NicolasMahe) [inline frontend/components/admin/allowance.vue]: why allowance has add, edit and delete function?
Should only have one "edit" function to change the allowance.
And anyway, this PR should be about Distribution Config store. No UI and even less UI about allowance.

File: frontend/store/allowance.js (hunk 2 above)
Comment (patjanawat) [reply frontend/store/allowance.js]: Done.

File: frontend/store/allowance.js (hunk 1 above)
Comment (patjanawat) [reply frontend/store/allowance.js]: Done.

File: frontend/store/distributionConfige.js (hunk 3 above)
Comment (patjanawat) [reply frontend/store/distributionConfige.js]: Done.

File: frontend/store/distributionConfige.js (hunk 4 above)
Comment (patjanawat) [reply frontend/store/distributionConfige.js]: Done.

File: frontend/store/distributionConfige.js (hunk 5 above)
Comment (patjanawat) [reply frontend/store/distributionConfige.js]: Done.

File: frontend/components/admin/allowance.vue (hunk 6 above)
Comment (patjanawat) [reply frontend/components/admin/allowance.vue]: Benz, he will fix it after.

**Comment by NicolasMahe** [comment]
Closing in favor of https://github.com/mesg-foundation/atd/pull/62

**Pull request closed without merging**